### PR TITLE
Compile switch to disable VFS

### PR DESCRIPTION
**Pull request by aykevl**
Based on https://github.com/tralamazza/micropython/issues/96#issuecomment-335299805

Depends on `--gc-sections` (#111) to properly clean up all oofatfs references (otherwise gives many 'undefined reference' linker errors).

You can disable VFS with a single switch in e.g. mpconfigboard.h:

    #define MICROPY_VFS (0)

@glennrub I feel like I should credit you in here somehow, it's mostly based on what you did.

### CODE REVIEW

**Comment by glennrub** [comment]
I would like to test this using SPI mounted SD-card. Specifically if it affects the mountsd example: https://github.com/tralamazza/micropython/blob/master/ports/nrf/examples/mountsd.py#L30

**Comment by aykevl** [comment]
As the `os.mount` function is removed, I'm pretty sure it won't work (and cannot work, that's the whole point of removing VFS/FAT).
It may be possible to keep SD card access without filesystem (as a raw block device), but cannot test it as I don't have an SD card holder.

File: ports/nrf/mpconfigport.h [hunk 1]
```diff
@@ -263,10 +267,16 @@ extern const struct _mp_obj_module_t ble_module;
     { MP_ROM_QSTR(MP_QSTR_os), MP_ROM_PTR(&mp_module_uos) }, \
     { MP_ROM_QSTR(MP_QSTR_time), MP_ROM_PTR(&mp_module_utime) }, \
 
+#if MICROPY_VFS
+#define BUILTIN_OPEN { MP_ROM_QSTR(MP_QSTR_open), MP_ROM_PTR(&mp_builtin_open_obj) },
+#else
```
Comment (glennrub) [inline ports/nrf/mpconfigport.h]: I took a look at the minimal port. I guess this is one of the methods that are expected to be found on all platforms. Should we consider to keep the open() compiled in, and add a MICROPY_VFS define in builtin_open.c? The builtin_open.c is not in the Makefile source file list at the moment, but would serve this purpose of selecting between FAT or dummy implementation.

Comment (aykevl) [reply ports/nrf/mpconfigport.h]: Any idea why `open` would exist on platforms without file support? Just for consistency? And what about `os.listdir()`?
Sure, we can keep it in using a `MICROPY_VFS` guard. Note that I would like to add microbit filesystem without VFS support soon (like on the microbit itself), so `open()` may actually be useful on all platforms.

Comment (aykevl) [reply ports/nrf/mpconfigport.h]: Added a stub `open()` call.
I've also removed `builtin_open.c`, as it didn't seem to be referenced anywhere.

File: ports/nrf/main.c [hunk 2]
```diff
@@ -213,6 +214,22 @@ pin_init0();
...
+    return MP_IMPORT_STAT_NO_EXIST;
+}
+
+STATIC mp_obj_t mp_builtin_open(size_t n_args, const mp_obj_t *args, mp_map_t *kwargs) {
+    mp_raise_OSError(MP_EPERM);
+}
+MP_DEFINE_CONST_FUN_OBJ_KW(mp_builtin_open_obj, 1, mp_builtin_open);
+#endif
+
+
```
Comment (glennrub) [inline ports/nrf/main.c]: for consistency, one newline could be removed.

Comment (aykevl) [reply ports/nrf/main.c]: done

**Comment by glennrub** [comment]
I have tested the disabled mode, and it working as expected. Also, a big thanks for removing the unused file :-)

**Comment by aykevl** [comment]
thanks for merging!

**Pull request merged**